crates/turborepo-lib/src/run/mod.rs

```diff
@@ -514,6 +524,9 @@ impl<'a> Run<'a> {
             true,
             self.processes.clone(),
             &self.base.repo_root,
+            // TODO: this is only needed for full execution, figure out better way to model this
+            // not affecting a dry run
+            EnvironmentVariableMap::default(),
```

Comment:
is the only concern perf of a dry run?

Reply:
Not so much perf, `EnvironmentVariableMap` is just a wrapper around `HashMap` and `HashMap::default` (which is just `HashMap::new`) is pretty cheap as it doesn't actually allocate any memory.

I'm bothered by needing to provide a structure that isn't used at all in this codepath.